src/ast.ts

```diff
@@ -326,20 +336,62 @@ export class AST {
   }
 }
 
+export type Pos = {
+  line: number;
+  ch: number;
+}
```

Comment:
Here is an example of a place that could maybe be refactored to be more semantically accurate. I created this `Pos` type since objects like this get passed around all over the place, but they look almost identical to a type from codemirror: `CodeMirror.Position`. So the question is, semantically, are we really intending to use codemirror's "position" objects everywhere or is it that we just so happened to choose the same data model for something that is semantically different from codemirror's "position" objects?

Reply:
Oh no, we are _100%_ using the semantics of CM's type. However, it looks like CM6 has reverted to using character index as their unit of measurement, so we may need to maintain our own type internally.